Alignment/affinealignment.cpp

```diff
@@ -159,6 +179,8 @@ AlignedIndices getAffineAlignedIndices(AffineAlignObj &affineAlignObj){
 //    std::cout << std::endl;
 //    std::cout << "************************" << std::endl;
 
+    // TODO: appending to the front is highly inefficient for std::vector - it causes memory re-allocation for every event!
+    // Instead: use push_back to append to back, and then std::reverse!
     alignedIdx.indexA_aligned.insert(alignedIdx.indexA_aligned.begin(), ROW_IDX);
```

Comment:
e.g. replace this with 
```suggestion
     alignedIdx.indexA_aligned.push_back(ROW_IDX);
```

which is efficient

Reply:
see also here https://en.cppreference.com/w/cpp/container/vector/insert

```
Complexity
1-2) Constant plus linear in the distance between pos and end of the container.
```

so if you insert at the beginning then the complexity is equal to the number of elements in the container

Reply:
Read about it and changed the code in the DIAlignR package.